### PR TITLE
PAYSHIP-2500 PayPal SDK spike

### PR DESCRIPTION
**Pull request by L3RAZ**

### CODE REVIEW

File: _dev/js/front/src/components/common/payment-option.component.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ export class PaymentOptionComponent extends BaseComponent {
         : this.$('funding-source.name.default');
     let element = Array.prototype.slice
       .call(this.data.HTMLElementContainer.querySelectorAll('*'))
-      .find(item => item.innerHTML.trim() === label.trim());
+      .find(item => item.innerText.trim() === label.trim());
```
Comment (L3RAZ) [inline _dev/js/front/src/components/common/payment-option.component.js:92]: Check on 1.6

File: _dev/js/front/src/components/common/paypal-sdk.component.js [hunk 2]
```diff
@@ -29,44 +30,50 @@ export class PayPalSdkComponent extends BaseComponent {
...
+        .catch((error) => {
+          console.error("failed to load the PayPal JS SDK script", error);
+          reject();
+        });
     });
 
     return this;
   }
+
+  reload() {
```
Comment (L3RAZ) [inline _dev/js/front/src/components/common/paypal-sdk.component.js:76]: Check if reload is possible

File: _dev/js/front/src/service/paypal.service.js [hunk 3]
```diff
@@ -329,6 +329,7 @@ export class PayPalService extends BaseClass {
       ...(this.configPayPal.payLaterOfferMessageCustomization || {}),
       ...(window.ps_checkout.payLaterOfferMessageCustomization || {})
     };
+    console.log('SDK',this.sdk);
```
Comment (L3RAZ) [inline _dev/js/front/src/service/paypal.service.js:332]: Remove

File: src/Builder/PayPalSdkConfiguration/PayPalSdkConfigurationBuilder.php [hunk 4]
```diff
@@ -18,18 +18,19 @@
...
 use PrestaShop\Module\PrestashopCheckout\PayPal\PayPalPayLaterConfiguration;
+use PrestaShop\Module\PrestashopCheckout\ShopContext;
 
 /**
  * Build sdk link
  */
-class PayPalSdkLinkBuilder
+class PayPalSdkConfigurationBuilder
 {
     const BASE_LINK = 'https://www.paypal.com/sdk/js';
```
Comment (L3RAZ) [inline src/Builder/PayPalSdkConfiguration/PayPalSdkConfigurationBuilder.php:35]: Not needed

File: views/templates/hook/displayPaymentByBinaries.tpl [hunk 5]
```diff
@@ -18,7 +18,7 @@
  *}
 
 {foreach from=$paymentOptions item="fundingSource"}
-  <section class="js-payment-binary js-payment-ps_checkout js-payment-ps_checkout-{$fundingSource} disabled">
+  <section class="js-payment-binary js-payment-ps_checkout js-payment-ps_checkout-{$fundingSource} mt-1 disabled">
     <p class="alert alert-warning accept-cgv">{l s='You must accept the terms and conditions to be able to process your order.' mod='ps_checkout'}</p>
```
Comment (L3RAZ) [inline views/templates/hook/displayPaymentByBinaries.tpl:22]: Check this because the alert doesn't disappear

**Review by Matt75 — APPROVED** [review]

**Pull request merged**